CrazyCanvas/Source/World/Player/PlayerActionSystem.cpp

```diff
@@ -94,24 +94,42 @@ bool PlayerActionSystem::OnKeyPressed(const KeyPressedEvent& event)
 	return false;
 }
 
-void PlayerActionSystem::ComputeVelocity(const glm::quat& rotation, int8 deltaForward, int8 deltaLeft, glm::vec3& result)
+void PlayerActionSystem::ComputeVelocity(const glm::quat& rotation, const glm::i8vec3& deltaAction, glm::vec3& result)
 {
-	if (!Match::HasBegun() || (deltaForward == 0 && deltaLeft == 0))
+	bool horizontalMovement = deltaAction.x != 0 || deltaAction.z != 0;
+	bool verticalMovement = deltaAction.y != 0;
+	bool anyMovement = horizontalMovement || verticalMovement;
```

Comment:
I think this is a pretty way of presenting the cases that decides behavior. Gold Star ✨